### PR TITLE
SG-34551 validation for python3, so display_name now is only decoded …

### PR DESCRIPTION
**Pull request by eduardoChaucaGallegos**
Now prefix/suffix if display_name property exists is not added

### CODE REVIEW

File: shotgun_api3/shotgun.py [hunk 1]
```diff
@@ -2587,7 +2587,10 @@ def _upload_to_sg(self, entity_type, entity_id, path, field_name, display_name,
             url = urllib.parse.urlunparse((self.config.scheme, self.config.server,
                                            "/upload/upload_file", None, None, None))
             if display_name is None:
-                display_name = os.path.basename(path)
+                if six.PY3:
```
Comment (julien-lang) [inline shotgun_api3/shotgun.py:2590]: Minor comment: I'd prefer having the opposite test: if Py2:... else.
So the specific is about Py2.

**Review by julien-lang — COMMENTED** [review]
Looks good. Are there any other fields that could have the same problem? 

Also is it possible to add unit tests?

**Comment by eduardochauca90** [comment]
@julien-lang  By the moment I couldn't see other fields with that potential problem in the scope of "_upload_to_sg" function because it is the only one field that previously was encoded to utf-8(encode("utf-8")) to assurance the behavior of non-ascii unicode and utf-8 string paths during upload.
Also I believe that is not necessary a new test for that new condition because that function and complete upload flow already has a unit tests. There exists a possibility to use another python2 "runner" in settings of CI/CD, but If I'm not wrong, we only will support python3 in the next weeks

File: tests/test_api.py [hunk 2]
```diff
@@ -310,6 +311,54 @@ def test_upload_download(self):
...
+    @patch('shotgun_api3.Shotgun._send_form')
+    def test_upload_to_sg(self, mock_send_form):
+        """
+        Upload an attachment tests for _upload_to_sg()
+        """
+        mock_send_form.method.assert_called_once()
+        mock_send_form.return_value = "1\n:123\nasd"
+        if 'localhost' in self.server_url:
+            print("upload / down tests skipped for localhost")
+            return
```
Comment (carlos-villavicencio-adsk) [inline tests/test_api.py:323]: Another option could be to use pytest's [skipif](https://docs.pytest.org/en/latest/how-to/skipping.html#id1) marker.

Not a blocker, just to let you know you have this option.

File: tests/test_api.py [hunk 3]
```diff
@@ -24,6 +24,7 @@
 import uuid
 import unittest
 from shotgun_api3.lib.six.moves import range, urllib
+import pytest
```
Comment (julien-lang) [inline tests/test_api.py:27]: I don't see how this import is required.

File: tests/test_api.py (hunk 2 above)
Comment (julien-lang) [reply tests/test_api.py:323]: Agreed. Either a decorator or just call [skipIf](https://docs.python.org/3/library/unittest.html#unittest.skipIf):

```suggestion
        self.skipIf(
            'localhost' in self.server_url,
            "upload / down tests skipped for localhost",
        )
```

**Review by carlos-villavicencio-adsk — APPROVED** [review]
LGTM!

File: shotgun_api3/shotgun.py [hunk 4]
```diff
@@ -2587,7 +2587,10 @@ def _upload_to_sg(self, entity_type, entity_id, path, field_name, display_name,
             url = urllib.parse.urlunparse((self.config.scheme, self.config.server,
                                            "/upload/upload_file", None, None, None))
             if display_name is None:
-                display_name = os.path.basename(path)
+                if six.PY2:
+                    display_name = os.path.basename(path)
+                else:
+                    display_name = os.path.basename(path.decode("utf-8"))
```
Comment (julien-lang) [inline shotgun_api3/shotgun.py:2593]: Can you 15 lines above. There is a big comment about unicode path and we encode path in utf-8... Looks like the opposite of what's done here. We might want to refactor/clean/review all that

**Review by carlos-villavicencio-adsk — APPROVED** [review]

**Review by julien-lang — APPROVED** [review]
LGTM!

**Pull request merged**